src/js/router.js

```diff
@@ -0,0 +1,90 @@
+// Base grade for express router modules...
+//
+// The express module will automatically attempt to wire in anything with this gradeName into its routing table.
+
+var fluid     = fluid || require('infusion');
+var namespace = "gpii.express.router";
+var router    = fluid.registerNamespace(namespace);
+
+router.createRouter = function(that) {
+    if (!that.options.config || !that.options.config.express) {
+        console.error("Can't instantiate router without a working config object.")
+        return null;
+    }
+
+    var express       = require("express");
+    that.applier.change("router", express.Router());
+
+    that.events.routerLoaded.fire();
+};
+
+router.loadMiddleware = function(that) {
+    if (!that.model.router) {
+        console.log("Can't wire in child modules unless I have a router module.");
+        return;
+    }
+
+    if (!that.options.path) {
+        console.log("Can't wire in child modules without knowing my own path.");
+        return;
+    }
+
+    if (that.options.components) {
+        // Any child components with the grade {gpii.express.router} will be wired in and then added to our routing
+        Object.keys(that.options.components).forEach(function(key) {
```

Comment:
I thought I could at least get middleware working with the new pattern, but it also has problems when the middleware is invoked from a nested router module.